pkg/common/telemetry/server/datastore/wrapper.go

```diff
@@ -0,0 +1,167 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	"github.com/spiffe/spire/pkg/server/plugin/datastore"
+)
+
+func WithMetrics(ds datastore.DataStore, metrics telemetry.Metrics) datastore.DataStore {
```

Comment:
I think that it would be nice to document this function, to make it clear what is the intended purpose and usage.